cluster-autoscaler/simulator/clustersnapshot/clustersnapshot.go

```diff
@@ -43,14 +43,16 @@ type ClusterSnapshot interface {
 	IsPVCUsedByPods(key string) bool
 
 	// Fork creates a fork of snapshot state. All modifications can later be reverted to moment of forking via Revert()
-	// Forking already forked snapshot is not allowed and will result with an error.
 	Fork() error
 	// Revert reverts snapshot state to moment of forking.
 	Revert() error
 	// Commit commits changes done after forking.
 	Commit() error
-	// Clear reset cluster snapshot to empty, unforked state
+	// Clear reset cluster snapshot to empty, unforked state.
 	Clear()
 }
 
 var errNodeNotFound = errors.New("node not found")
+
+// ForkLimit is a defautl fork limit for cluster cnapshot.
+const ForkLimit = 2
```

Comment:
Why impose an arbitrary limit?

Reply:
The main reason why there was a limitation to fork once previously is to ensure we don't accidentally fork twice.
The main reason why I added a fork limit is to control recursion depth of forking. Currently we limit fork to two time, so I used this value as a new default.

Reply:
I disagree handling this as a runtime error is the right approach. If we are concerned there are parts of the logic which could fork without committing/reverting, there should be unit tests ensuring this doesn't happen.

Reply:
Alternatively, we could modify the interface in a way that ensures forking without commiting/reverting is not possible at all. I actually did just that in one place already:
https://github.com/kubernetes/autoscaler/blob/a703c3f6cff1d54dc92378c5baad2c2cccdbbc8f/cluster-autoscaler/simulator/cluster.go#L198-L217

Reply:
Done